sql_app/crud.py

```diff
@@ -16,12 +16,26 @@ def get_movie(db: Session, movie_id: int):
 
 
 @lru_cache(maxsize=1000)
-def list_movies(db: Session, engine: str, page: int, num: int):
+def list_movies(db: HashableSession, engine: str, page: int, num: int):
     """
     Get List of Movies. With pagination and a total of `num` rows per query
     """
     return list(db.query(models.Movie).filter(models.Movie.engine == engine).limit(num).offset(num * (page-1)))
 
+@lru_cache(maxsize=1000)
+def search_movies(db: HashableSession, engine: str, query: str, page:int, num: int):
+    """
+    Search movies using fuzzy partial
+    """
+    ret =  list(db.query(models.Movie)\
+        .filter(\
+            models.Movie.name.ilike("%"+query+"%"),
```

Comment:
`ilike` will not get instances when the title is no fully correct. Say for instance searching `iron man` and the name of the movie is `ironman`, `ilike` will fail to get that.

Reply:
Yeah I get... but so far this is the only way we can query from what I've seen... I initially considered fuzzy matching, but couldn't get it to work with the sql Comparator

Reply:
I think we will have to return all the movies and run the search against that

Reply:
We won't. I've found something don't worry

Reply:
🙌🙌🙌

Reply:
The most accurate so far is ilike man... fuzzy partial search just flies all over the place